### PR TITLE
fix: background local test also via channel

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
fix: background local test also via channel

## Motivation and Context
the current implementation for `standalone` setups
was blocking the `perf drive`.

## How to test this PR?
`mc support perf drive alias/`  

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

File: cmd/notification.go [hunk 1]
```diff
@@ -1753,19 +1754,28 @@ func (sys *NotificationSys) DriveSpeedTest(ctx context.Context, opts madmin.Driv
...
-		wg.Wait()
-		close(ch)
+		defer wg.Done()
+		select {
+		case <-ctx.Done():
+		case ch <- driveSpeedTest(ctx, opts):
+		}
 	}()
 
+	wg.Wait()
```
Comment (vadmeste) [inline cmd/notification.go]: DriveSpeedTest() is an async function, it is supposed to return 'ch' and the caller consumes it. We should not wait here

**Review by vadmeste — CHANGES_REQUESTED** [review]

File: cmd/notification.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/notification.go]: yes my mistake, making the change @vadmeste

**Review by krisis — APPROVED** [review]

**Review by vadmeste — APPROVED** [review]
LGTM

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Pull request merged**